pkg/registry/apis/provisioning/register.go

```diff
@@ -583,8 +586,15 @@ func (b *APIBuilder) GetPostStartHooks() (map[string]genericapiserver.PostStartH
 			commenter := pullrequest.NewCommenter()
 			pullRequestWorker := pullrequest.NewPullRequestWorker(evaluator, commenter)
 
-			driver := jobs.NewJobDriver(time.Second*28, time.Second*30, time.Second*30, b.jobs, b, b.jobHistory,
+			driver, err := jobs.NewJobDriver(
+				jobTimeout,                  // Max time for each job
+				jobTimeout+(30*time.Second), // Cleanup any checked out jobs
```

Comment:
With this we wouldn't claim for 15.5 minutes after restart if it was killed in the middle. I would keep it to 30 seconds because it will only work now for restart. I would add a FIXME in the code to mention that the claiming logic must be reimplemented for multi-tenant (and add a Github Issue) for this in our board.

Reply:
aaahh -- we can use the fact that it is a single thread, and know that we are only cleaning up jobs when we are not also running them

Reply:
https://github.com/grafana/git-ui-sync-project/issues/218